deploy/internal/pod-agent.yaml

```
@@ -0,0 +1,41 @@
+apiVersion: v1
+kind: Pod
+metadata:
+  labels:
+    app: noobaa
+  name: noobaa-agent
+spec:
+  containers:
+    - name: noobaa-agent
+      image: NOOBAA_CORE_IMAGE
+      imagePullPolicy: IfNotPresent
+      resources:
+        # https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/
+        requests:
+          cpu: "100m"
+          memory: "200Mi"
+        limits:
+          cpu: "100m"
+          memory: "200Mi"
+      env:
+        # Insert the relevant config for the current agent
+        - name: CONTAINER_PLATFORM
+          value: KUBERNETES
+        - name: AGENT_CONFIG
+          value: "AGENT_CONFIG_VALUE"
```

also use `valueFrom` and `secretKeyRef` inside the pod. that way you don't expose the secret in the pod CR.

```suggestion
        - name: AGENT_CONFIG
           valueFrom:
             secretKeyRef:
               name: secret-name
               key: AGENT_CONFIG
```

---

using secret - fixed